code/game/turfs/simulated/walls_reinforced.dm

```diff
@@ -293,4 +311,13 @@
 		clear_smooth_overlays()
 	else
 		smooth = SMOOTH_TRUE
-		icon_state = ""
\ No newline at end of file
+		icon_state = ""
+
+/turf/simulated/wall/r_wall/coated			//Coated for the heat resistance
+	name = "coated reinforced wall"
+	desc = "A huge chunk of reinforced metal used to seperate rooms. It seems to have additional plating to protect against heat."
+	icon = 'icons/turf/walls/reinforced_wall.dmi'
+	icon_state = "r_wall"
+	max_temperature = INFINITY

```

Comment:
I don't think adding overrides only for this one type of wall would be a good idea rather than having it work off of a variable of any wall.